File: Java/Seller.java
import java.util.ArrayList;
import java.util.List;

public class Seller extends Person {
    private List<Plant> plantProductList;
    private List<Farmer> farmerList;

    // Default constructor
    public Seller() {
        super();
        this.plantProductList = new ArrayList<>();
        this.farmerList = new ArrayList<>();
    }

    // Parameterized constructor
    public Seller(int idPerson, String name, String address) {
        super(idPerson, name, address);
        this.plantProductList = new ArrayList<>();
        this.farmerList = new ArrayList<>();
    }

    // ===== Getter methods =====
    public List<Plant> getPlantProductList() {
        return this.plantProductList;
    }

    public List<Farmer> getFarmerList() {
        return this.farmerList;
    }

    // ===== Setter methods =====
    public void setPlantProductList(List<Plant> daftar) {
        this.plantProductList = daftar;
    }

    public void setFarmerList(List<Farmer> daftar) {
        this.farmerList = daftar;
    }

    // Tambah Plant ke daftar produk
    public void addPlantProduct(Plant plant) {
        this.plantProductList.add(plant);
    }

    // Tambah Farmer ke daftar mitra
    public void addFarmer(Farmer farmer) {
        this.farmerList.add(farmer);
    }

    // Display
    public void displaySeller(int indent) {
        String sp = " ".repeat(indent);
        System.out.println(sp + "Seller ID: " + getIdPerson());
        System.out.println(sp + "Name     : " + getName());
        System.out.println(sp + "Address  : " + getAddress());
        System.out.println();

        if (!plantProductList.isEmpty()) {
            System.out.println(sp + "    Products:");
            for (Plant plant : plantProductList) {
                plant.displayPlant(indent + 8);
            }
        }

        if (!farmerList.isEmpty()) {
            System.out.println(sp + "    Farmers:");
            for (Farmer farmer : farmerList) {
                farmer.displayFarmer(indent + 8);
            }
        }
    }
}

File: Python/Seller.py
from Person import Person
from Plant import Plant
from Farmer import Farmer

class Seller(Person):
    def __init__(self, id_person=0, name="", address=""):
        super().__init__(id_person, name, address)
        self.plant_product_list = []
        self.farmer_list = []

    # Getter
    def get_plant_product_list(self):
        return self.plant_product_list

    def get_farmer_list(self):
        return self.farmer_list

    # Setter
    def set_plant_product_list(self, daftar):
        self.plant_product_list = daftar

    def set_farmer_list(self, daftar):
        self.farmer_list = daftar

    # Tambah Plant ke daftar produk
    def add_plant_product(self, plant: Plant):
        self.plant_product_list.append(plant)

    # Tambah Farmer ke daftar mitra
    def add_farmer(self, farmer: Farmer):
        self.farmer_list.append(farmer)

    # Display Seller
    def display_seller(self, indent=0):
        sp = " " * indent
        print(f"{sp}Seller ID: {self.get_id_person()}")
        print(f"{sp}Name     : {self.get_name()}")
        print(f"{sp}Address  : {self.get_address()}\n")

        if self.plant_product_list:
            print(f"{sp}    Products:")
            for plant in self.plant_product_list:
                plant.display_plant(indent + 8)

        if self.farmer_list:
            print(f"{sp}    Farmers:")
            for farmer in self.farmer_list:
                farmer.display_farmer(indent + 8)

    def __del__(self):
        # Destructor opsional
        pass
